### ISSUE TITLE
No `traits` on any events after updating from version 1.5.0 to 2.2.2

### ISSUE THREAD

**Issue by Wiingaard** [segmentio/analytics-react-native#578]
After updating from version `1.5.0` to version `2.2.2`, `traits` no longer show up the the `context.traits` of any event.

When inspecting `track` events in the online debugger `context.traits` is an empty object on every event, even though identify has been called prior to the events.

- `analytics-react-native` version: 2.2.2
- Integrations versions: `@segment/analytics-react-native-plugin-amplitude-session` version `0.2.1`
- React Native version: `0.65.2`
- iOS or Android or both? Both

## Steps to reproduce

In global scope I initialize the client like so:

```
import {createClient} from '@segment/analytics-react-native'
import {AmplitudeSessionPlugin} from '@segment/analytics-react-native-plugin-amplitude-session'
import config from './config'

const segmentKey = config.segmentId

const client = createClient({
    writeKey: segmentKey,
    trackAppLifecycleEvents: true,
    collectDeviceId: true,
    trackDeepLinks: true
})

client.add({plugin: new AmplitudeSessionPlugin()})

export default client
```

Once the users successfully signs in, I call identify like so:

```
import Analytics from 'src/lib/analytics'
...
Analytics.identify(userId, {firstName, lastName})
```

When the user performs some action, I track it events like so:

```
import Analytics from 'src/lib/analytics'
...
Analytics.track('Banner Clicked', {
    type: 'missing_details'
})
```

## Behavior

When inspecting the online debugger I see the `IDENTIFY` log like so (cut down for privacy and brevity):

```
{
  "anonymousId": "kM_gljge4syZRiAqGiYSq",
  "context": {
    ...
    "traits": {}
  },
  ...
  "traits": {
    "firstName": "M*** A***",
    "lastName": "W***",
  },
  ...
  "type": "identify",
  "userId": "ca5ed865-0761-4181-ae7d-c26f2a12d4bd",
  ...
}
```
Notice: `context.traits` is empty but the `"traits"` property is set.

On a subsequent `TRACK` event, I still don't see `context.traits`:

```
{
  "anonymousId": "kM_gljge4syZRiAqGiYSq",
  "context": {
    ...
    "traits": {}
  },
  "event": "Banner Clicked",
  ...
  "properties": {
    "type": "missing_details"
  },
  ...
  "type": "track",
  "userId": "ca5ed865-0761-4181-ae7d-c26f2a12d4bd",
  ...
}
```

**Why is `context.traits` empty even though I did the identify prior to the tracking call?**

I experience this behavior both when running on Android and iOS.

Another comment is that this used to work before I updated the library from version `1.5.0` to version `2.2.2`. When looking at tracking and identify events sent from app versions that are still running the old SDK version, I see that `context.traits` are filled as expected. I didn't change anything on the tracking or identify events during the update. All I did was following the [migration guide](https://segment.com/docs/connections/sources/catalog/libraries/mobile/react-native/migration/) which updated dependencies and moved the setup to global scope. 

Any help would be much appreciated.. We're doing this update because of some notice that the android library linked to the version we're currently using is non-compliant with their User Data policy. Google is prompting us to update to version `4.10.1` or we'll not be able to submit new builds, so we're a bit in a hurry here..

**Comment by Wiingaard** [comment]
I'm able to reproduce this issue on a clean installed react native app. See source code [here](https://github.com/Wiingaard/DebugSegment/tree/master)..

This clean app is running the latest react-native 0.69.0. So that doesn't seem to be the cause either..

Notice timestamps of track and identify. Identify happens first, yet still no traits on the tracking event.
<img width="1380" alt="Screenshot 2022-06-24 at 12 23 31" src="https://user-images.githubusercontent.com/8324312/175516767-96e16e33-ef0e-4846-bb3a-53d8e2e974f5.png">

**Comment by oscb** [comment]
Thanks the report! This is confirmed a bug in v2. 
We'll start working on a fix due on the next release, should be going out on Monday at the latest.

**Comment by oscb** [comment]
This was released in: https://github.com/segmentio/analytics-react-native/releases/tag/%40segment%2Fanalytics-react-native-v2.3.1

**Comment by Wiingaard** [comment]
Thanks a lot for the quick response and fix, Oscar! 🙌 

I've upgraded to version 2.3.1 and verified that the traits are now sent on tracking events. It seems however that they have now been added directly to the BaseEvent instead of to the `context`... 

It seem that traits are now sent along as parameters on each tracking event, form the debugger I see:
```
curl https://api.segment.io/v1/screen \
  -d anonymousId=kM_gljge4syZRiAqGiYSq \
  -d messageId=3YFSDmTEhYUeoqLx5frVm \
  -d name=BannerClicked \
  -d originalTimestamp=2022-06-27T08:39:46.605Z \
  -d receivedAt=2022-06-27T08:39:55.805Z \
  -d sentAt=2022-06-27T08:39:53.507Z \
  -d "traits[firstName]=M*** A***" \
  -d traits[lastName]=W*** \
  -d userId=ca5ed865-0761-4181-ae7d-c26f2a12d4bd \
  -d writeKey=xxx
```
Whereas on events from the old SDK, they don't seem to be sent as parameters, but rather magically be attached `context.traits` when inspecting the `Raw` output. 
I attempted overwriting the `applyRawEventData` that you edited in the fix from #581 like so:

```
export const applyRawEventData = (
  event: SegmentEvent,
  userInfo: UserInfoState
): SegmentEvent => {
  return {
    ...event,
    anonymousId: userInfo.anonymousId,
    messageId: getUUID(),
    timestamp: new Date().toISOString(),
    integrations: event.integrations ?? {},
    context: {
      ...event.context,
      traits: userInfo.traits ?? {}
    },
    userId:
      isAliasEvent(event) || isIdentifyEvent(event)
        ? event.userId
        : userInfo.userId
  };
};
```

But that also doesn't make the traits appear in the context in the debugger..

**Comment by Wiingaard** [comment]
This issue got closed, so I've opened a new issue [here](https://github.com/segmentio/analytics-react-native/issues/586) about the discrepancy between the new and old library versions..
